docs/book/src/quinn/data-transfer.md

```diff
@@ -0,0 +1,172 @@
+# Data Transfer
+
+In the [previous chapter](set-up-connection.md) we characterized how to set up an [Endpoint][Endpoint]
+and then get access to a [NewConnection][NewConnection] instance.
+Now we will continue with the subject of sending data over this connection.
+
+## Multiplexing
+
+A QUIC stream can be compared to a TCP stream, eventhough if it is based on UDP. 
+The big difference is that you are not limited to a single stream. 
+You can open multiple streams between two peers, this is also called multiplexing.
+
+Stream multiplexing can have a significant positive effect on application performance if the resources allocated to streams are properly prioritized.
+Currently multiplexing it is used in a protocol like HTTP/2, but unlike HTTP multiplexing, QUIC does not automatically multiplex.
+This is an advantage because the user gets full control over the multiplexing. 
+Another difference compared to a TCP stream is that QUIC is not limited to one kind of stream but can open both bidirectional and unidirectional streams.
+ 
+## Stream Types
+
+Quinn offers three ways to send your data. 
+Two stream-based and one message-based.
+
+| Type | Description | Reference |
+| :----- | :----- | :----- |
+| **Bidirectional Stream** | two way stream communication. | see [open_bi][open_bi] |
+| **Unidirectional Stream** | one way stream communication. | see [open_uni][open_uni] |
+| **Unreliable Messaging** | message based unreliable communication. | see [send_datagram][send_datagram] |
+
+## How to Use
+
+You can open a new stream or read from an existing stream.
+New streams can be created with the methods [open_bi][open_bi], [open_uni][open_uni] from [Connection][Connection] type.
+An instance of this type is found in the [connection][connection] field of [NewConnection][NewConnection]. 
+This [NewConnection][NewConnection] stores existing streams as well. Lets look at some examples:
+
+*Iterate over various opened streams*
+
+```rust
+async fn iterate_streams(mut connection: NewConnection) -> anyhow::Result<()> {
+    // Iterate unidirectional streams with only the receiving side.
+    while let Some(Ok(recv)) = connection.uni_streams.next().await { }
+    // Iterate bidirectional streams with both sent and receiving side.
+    while let Some(Ok((sent, recv))) = connection.bi_streams.next().await { }
+    // Iterate arrived datagrams.
+    while let Some(Ok(bytes)) = connection.datagrams.next().await { }
+
+    Ok(())
+}
```

Comment:
"opened" doesn't seem like the best term here, let's use "incoming" instead to make it clear that these were initiated by the peer?

Reply:
I will just remove this word and leave it at 'iterate over various streams'. Incoming is may not be correct because it can be initiated from the endpoint itself.